### PR TITLE
[Merged by Bors] - Fix bug in StartPost

### PR DESCRIPTION
**Pull request by lrettig**
case statement was missing fallthrough. Updates tests accordingly.

## Motivation
Spotted while working on #1700 (but probably unrelated).

## Changes
Fix case statement

## Test Plan
Updated tests accordingly.

### CODE REVIEW

File: activation/activation.go [hunk 1]
```diff
@@ -223,6 +223,7 @@ func (b *Builder) StartPost(rewardAddress types.Address, dataDir string, space u
 		case InitDone:
 			return fmt.Errorf("already initialized")
 		case InitInProgress:
+			fallthrough
 		default:
```
Comment (noamnelke) [inline activation/activation.go]: I think that it's better to remove the `default:` clause here, than add a `fallthrough`. There aren't any other values that `b.initStatus` can have that aren't explicitly covered by the `switch` (it won't get to the switch if it's `InitIdle`).

```suggestion
```

**Review by noamnelke — CHANGES_REQUESTED** [review]
Good catch!

Sorry for not getting to this sooner (you can ping me on slack next time if I miss PRs).

**Review by noamnelke — APPROVED** [review]

**Comment by avive** [comment]
Can someone please check if this fix also fixed #1879 - there are line numbers in the go code in the stack trace for this panic in the screen shot. @lrettig @noamnelke - I got this panic few times on my smesher.

**Comment by noamnelke** [comment]
I'm not 100% sure, but from quickly trying to follow the code (not all the way through) I think a `panic` is plausible.

However, in order for this to cause a `panic` a restart of the node during PoST initialization is required. Since you said you can reproduce it without a restart, at the very least there's another cause for that issue.

**Comment by lrettig** [comment]
bors merge

**Pull request closed without merging**